sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractFileCacheSensor.java

```diff
@@ -70,7 +77,10 @@ public void execute(SensorContext context) {
 
     LOG.debug("Incremental PR analysis: Preparing to upload file hashes.");
     var fileSystem = context.fileSystem();
-    fileSystem.inputFiles(fileSystem.predicates().hasLanguage(language.getKey())).forEach(inputFile -> {
+    var predicates = fileSystem.predicates();
+    var filePredicates = Arrays.stream(additionalSupportedExtensions()).map(extension -> fileSystem.predicates().hasExtension(extension)).collect(Collectors.toList());
+    filePredicates.add(predicates.hasLanguage(language.getKey()));
+    fileSystem.inputFiles(predicates.or(filePredicates)).forEach(inputFile -> {
```

Comment:
I do not fully understand how these predicates work. What does `fileSystem.predicates()` return exactly?
`predicates.or(filePredicates)` looks like it combines `predicates` with `filePredicates`, which would be a significant change in behavior to what we had before. But maybe `or` does only combine the predicates passed as an argument?

Reply:
It returns an object for creating file predicates (essentially a factory). Not sure how to rename it? `predicateFactory`?
`or()` returns a predicate that's `filePredicates[0] | filePredicates[1] | ...`